### PR TITLE
feat/HttpParser

### PR DESCRIPTION
**Pull request by beurie-v**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -93,8 +100,8 @@ set_target_properties(module_server
 enable_testing()
 #test executable
 add_executable(zia_httpd_server_test
-        ${PROJECT_SOURCE_DIR}/test/conf_test.cc
-        #[[add test source files here]])
+        ${PROJECT_SOURCE_DIR}/test/test.cc
```
Comment (MickaToast) [inline CMakeLists.txt]: This file doesn't exist anymore, that's why your CIs fails.

File: src/core/HttpParser.cc [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+	default:
+		res.version = zia::api::http::Version::unknown;
+		return true;
+		break;
+	}
+	return false;
+}
+
+bool
+HttpParser::_getHeaders(std::string &line, zia::api::HttpRequest &res) const {
```
Comment (MickaToast) [inline src/core/HttpParser.cc]: Could you set key to lowercase?

File: src/core/HttpParser.cc [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+		{ { "OPTIONS", zia::api::http::Method::options },
+		{ "GET", zia::api::http::Method::get },
+		{ "HEAD", zia::api::http::Method::head },
+		{ "POST", zia::api::http::Method::post },
+		{ "PUT", zia::api::http::Method::put },
+		{ "DELETE", zia::api::http::Method::delete_ },
+		{ "TRACE", zia::api::http::Method::trace },
+		{ "CONNECT", zia::api::http::Method::connect } };
+
+	for (std::pair<std::string, zia::api::http::Method> current : methodStringMap) {
```
Comment (MickaToast) [inline src/core/HttpParser.cc]: Use reference?

File: src/core/HttpParser.cc [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+	zia::api::HttpRequest       res;
+
+	res.method = zia::api::http::Method::unknown;
+	res.version = zia::api::http::Version::unknown;
+	res.uri = "";
+	
+	std::vector<std::string>    lines = this->_splitString(req, '\n');
+	
+	if (lines.empty()) {
+		return std::pair<zia::api::HttpRequest, bool>(res, true);
```
Comment (MickaToast) [inline src/core/HttpParser.cc]: Prefer using make_pair

File: src/core/HttpParser.cc [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+	}
+
+	std::string                 head = lines.front();
+
+	lines.erase(lines.begin());
+
+    return std::pair<zia::api::HttpRequest, bool>(res, this->_parseFirstLine(head, res) ||
```
Comment (MickaToast) [inline src/core/HttpParser.cc]: Prefer using make_pair

File: src/core/HttpParser.cc [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+	if (values.size() != 2) {
+		return true;
+	}
+	this->_epurStr(values[1]);
+	auto it = res.headers.find(values[0]);
+
+	if (it != res.headers.end()) {
+		res.headers[values[0]] += ", " + values[1];
+	} else {
+		res.headers.insert(std::pair<std::string, std::string>(values[0], values[1]));
```
Comment (MickaToast) [inline src/core/HttpParser.cc]: Prefer using make_pair

File: src/core/HttpParser.cc (hunk 4 above)
Comment (beurie-v) [reply src/core/HttpParser.cc]: Should be fixed

File: src/core/HttpParser.cc (hunk 3 above)
Comment (beurie-v) [reply src/core/HttpParser.cc]: Should be fixed

File: src/core/HttpParser.cc (hunk 5 above)
Comment (beurie-v) [reply src/core/HttpParser.cc]: Should be fixed

File: src/core/HttpParser.cc (hunk 6 above)
Comment (beurie-v) [reply src/core/HttpParser.cc]: Should be fixed

**Pull request merged**